app/api/v2/handlers/payload_api.py

```diff
@@ -22,22 +35,104 @@ def __init__(self, services):
     @aiohttp_apispec.response_schema(PayloadSchema(),
                                      description='Returns a list of all payloads in PayloadSchema format.')
     async def get_payloads(self, request: web.Request):
+        sort: bool = request['querystring'].get('sort')
+        exclude_plugins: bool = request['querystring'].get('exclude_plugins')
+        add_path: bool = request['querystring'].get('add_path')
+
         cwd = pathlib.Path.cwd()
         payload_dirs = [cwd / 'data' / 'payloads']
-        payload_dirs.extend(cwd / 'plugins' / plugin.name / 'payloads'
-                            for plugin in await self.data_svc.locate('plugins') if plugin.enabled)
+
+        if not exclude_plugins:
+            payload_dirs.extend(cwd / 'plugins' / plugin.name / 'payloads'
+                                for plugin in await self.data_svc.locate('plugins') if plugin.enabled)
+
         payloads = {
-            self.file_svc.remove_xored_extension(p.name)
+            str(p.parent.relative_to(cwd) / self.file_svc.remove_xored_extension(p.name))
+            if add_path
+            else self.file_svc.remove_xored_extension(p.name)
             for p in itertools.chain.from_iterable(p_dir.glob('[!.]*') for p_dir in payload_dirs)
             if p.is_file()
         }
 
-        return web.json_response(list(payloads))
+        payloads = list(payloads)
+        if sort:
+            payloads.sort()
+
+        return web.json_response(payloads)
+
+    @aiohttp_apispec.docs(
+        tags=['payloads'],
+        summary='Create a payload',
+        description='Uploads a payload.')
+    @aiohttp_apispec.form_schema(PayloadCreateRequestSchema)
+    @aiohttp_apispec.response_schema(
+        PayloadSchema(),
+        description="The created payload in a list in PayloadSchema format (with name changed in case of a duplicate).")
+    async def post_payloads(self, request: web.Request):
+        # As aiohttp_apispec.form_schema already calls request.multipart(),
+        # accessing the file using the prefilled request["form"] dictionary.
+        field_field: web.FileField = request["form"]["payload"]
+
+        file_name_candidate: str = field_field.filename
+        file_path: str = os.path.join('data/payloads/', file_name_candidate)
+        suffix: int = 1
+
+        # Generating a file suffix in the case it already exist.
+        while os.path.exists(file_path):
+            file_name_candidate = f"{pathlib.Path(field_field.filename).stem}_" \
+                                  f"{suffix}{pathlib.Path(field_field.filename).suffix}"
+            file_path = os.path.join('data/payloads/', file_name_candidate)
+            suffix += 1
+
+        file_name: str = file_name_candidate
+
+        # The field_field.file is of type IOBase: It uses blocking methods.
+        # Putting blocking code into a dedicated method and thread...
+        def save_file(target_file_path: str, io_base_src: IOBase):
```

Comment:
pull out into another class sub method.

Reply:
Done. I used a `@staticmethod` (no `cls` parameter) and used `__` prefix for private access (convention).